ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/VertxRequestTransmitter.java

```diff
@@ -68,17 +70,20 @@ public void sendRequest(
 
   private void handleException(final Throwable thrown) {
     LOG.error("Transmission failed", thrown);
-    vertx.executeBlocking(
-        future -> bodyHandler.handleFailure(thrown),
-        false,
-        res -> {
-          if (res.failed()) {
-            LOG.error("Reporting failure, failed", res.cause());
-          }
-        });
+    if (!responseHandled.getAndSet(true)) {
+      vertx.executeBlocking(
+          future -> bodyHandler.handleFailure(thrown),
+          false,
+          res -> {
+            if (res.failed()) {
+              LOG.error("Reporting failure, failed", res.cause());
+            }
+          });
+    }
   }
 
   private void handleResponse(final HttpClientResponse response) {
+    responseHandled.set(true);
```

Comment:
Do we also want to only handle responses once? Or is this only applicable for error responses?

Reply:
It appears the exception _can_ still fire after the response is handled, so want to prevent that - but I don't think secondary responses are possible.